trino/client.py

```diff
@@ -447,14 +448,15 @@ def transaction_id(self, value):
 
     @property
     def http_headers(self) -> Dict[str, str]:
-        headers = {}
```

Comment:
It might cause issues to use the default dict if the user tries to override any default header with a different case like `User-Agent` instead of `user-agent`, since the headers are case insensitive.

The requests is also [using](https://github.com/psf/requests/blob/v2.31.0/requests/utils.py#L898) this utility struct for creating the default header.